code/go/0chain.net/smartcontract/storagesc/readpool.go

```diff
@@ -368,41 +282,34 @@ func (ssc *StorageSmartContract) readPoolUnlock(t *transaction.Transaction,
 		req.PoolOwner = t.ClientID
 	}
 
-	isFunded, err := ssc.isFundedPool(t.ClientID, req.PoolID, balances)
-	if err != nil {
-		return "", common.NewError("read_pool_unlock_failed", err.Error())
-	}
-	if !isFunded {
-		return "", common.NewErrorf("read_pool_unlock_failed",
-			"%s did not fund pool %s", t.ClientID, req.PoolID)
-	}
-
-	var rp *readPool
-	if rp, err = ssc.getReadPool(req.PoolOwner, balances); err != nil {
-		return "", common.NewError("read_pool_unlock_failed", err.Error())
+	txValue := -(state.Balance(t.Value)) // subtract from read_pool balance
+	if err = ssc.modifyReadPoolBalance(t.ClientID, req.IsOwner, txValue, balances); err != nil {
```

Comment:
Somewhere we need to check there are enough funds in the readPool for the unlock and return an error if there are insufficient funds.